components/AddComment.tsx

```diff
@@ -1,34 +1,29 @@
 import { Button, Textarea } from '@geist-ui/react';
-import { useRouter } from 'next/router';
 import { useState } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
 
 import PivotalHandler from '../handlers/PivotalHandler';
-import { useAsync } from '../hooks';
+import { usePivotal } from '../hooks';
 import { editStory } from '../redux/actions/stories.actions';
-import { getApiKey } from '../redux/selectors/settings.selectors';
-import { Story, UrlParams } from '../redux/types';
+import { Story } from '../redux/types';
 
 interface AddCommentParams {
   story: Story;
 }
 
 const AddComment = ({ story }: AddCommentParams): JSX.Element => {
   const [comment, setComment] = useState<string>();
-  const apiKey = useSelector(getApiKey);
 
-  const router = useRouter();
   const dispatch = useDispatch();
-  const { id }: UrlParams = router.query;
 
-  const [{ isLoading }, addComment] = useAsync(async () => {
-    await PivotalHandler.addComment({ apiKey, projectId: id, storyId: story.id, text: comment });
+  const [{ isLoading }, addComment] = usePivotal(async ({ apiKey, projectId }) => {
```

Comment:
I'm still not super happy with this, but for now it will stick. I'll want to refactor these into actions or sagas at some point.